src/mscorlib/src/System/String.Manipulation.cs

```diff
@@ -6,14 +6,16 @@
 using System.Diagnostics;
 using System.Globalization;
 using System.Runtime.CompilerServices;
-using System.Runtime.InteropServices;
 using System.Text;
 using Internal.Runtime.CompilerServices;
+using System.Buffers;
 
 namespace System
 {
     public partial class String
     {
+        private const int StackallocStringLengthLimit = 512;
```

Comment:
512 feels a bit on the high side. Some of the methods allocate twice the size, and they will have 4kB+ stack frame. 

It may be worth looking at the right size for this - check what is the smallest size that helps microbenchmark results.